app/workers/step_by_step_scheduled_publish_worker.rb

```diff
@@ -7,13 +7,14 @@ class StepByStepScheduledPublishWorker
   end
 
   def perform(id)
-    step_nav = nil
+    step_nav = StepByStepPage.find_by(id: id)
 
-    StepByStepPage.transaction do
-      step_nav = StepByStepPage.lock.find_by(id: id)
-      StepNavPublisher.publish(step_nav)
-      step_nav.mark_as_published
-      generate_internal_change_note(step_nav)
+    if step_nav.scheduled_for_publishing?
```

Comment:
I see a potential problem with this:

1. Step by step is scheduled for publication on Friday
2. Step by step is unscheduled by the content designer (and from your commit message, it sounds like there's no way of clearing the sidekiq job queue at this point)
3. Step by step is scheduled for publication Saturday
4. When Friday comes around, `scheduled_for_publishing?` will be `true`, so we'll publish the step by step. But we'll be a day earlier than the content designer intended.

Reply:
Hmmm. Good point. I'll give this another thought.